aten/src/ATen/native/Indexing.cpp

```diff
@@ -272,7 +272,10 @@ Tensor & index_copy_(Tensor & self, int64_t dim, const Tensor & index, const Ten
   int64_t numIndices = index.numel();
   if (source.dim() == 0 && numIndices != 1) {
     AT_INDEX_ERROR("index_copy_(): When source is scalar, index should have one element (got ", numIndices, ")");
+  } else if (source.dim() != self.dim()) {
+    AT_INDEX_ERROR("index_copy_(): When source is not scalar, its dimensionality must match the dimensionality of the destination");
```

Comment:
It's nice to also list the dimensionalities so the user doesn't have to print out the tensors.
Something like:
```
index_copy_(): When source is not scalar, its dimensionality (3) must match the dimensionality of the destination (2)"
```

Reply:
also, the error message doesn't match what is checked -- the message says source is not a scalar, but this isn't actually checked.